### PR TITLE
add custom onChange directive to correct file uploader bug

### PR DESCRIPTION
**Pull request by frogfather**
Corrects a bug where uploading a file to the catalog via the file picker did not work.
https://issues.apache.org/jira/browse/BROOKLYN-624

Resolves an issue with the getCatalogItemUrl method, which checks for the existence of a string 'org.apache.brooklyn.api.location.Location' in item.supertypes to determine the format of the url:

- Adds a check for the existence of the property supertypes before calling 'includes' on it.
- Checks for the string in item.tags if the supertypes property is not present.

### CODE REVIEW

File: ui-modules/utils/catalog-uploader/catalog-uploader.js [hunk 1]
```diff
@@ -120,7 +121,9 @@ export function catalogUploaderDirective($compile, brooklynCatalogUploader) {
         };
 
         scope.getCatalogItemUrl = (item)=> {
-            return item.supertypes.includes('org.apache.brooklyn.api.location.Location')
+            let itemTraits = item.tags? item.tags.find(item => item.hasOwnProperty("traits")) : {"traits":[]};
+            return (item.supertypes ? item.supertypes : itemTraits.traits)
+                .includes('org.apache.brooklyn.api.location.Location')
```
Comment (tbouron) [inline ui-modules/utils/catalog-uploader/catalog-uploader.js]: Why is this change for? It doesn't look related to the file uploader

File: ui-modules/utils/catalog-uploader/catalog-uploader.js [hunk 2]
```diff
@@ -208,3 +211,20 @@ export function catalogUploaderService($q, catalogApi) {
...
+        link: function (scope, element, attrs) {
+            element.on('change', x => {
+                var onChangeHandler = scope.$eval(attrs.customOnChange);
+                onChangeHandler(x);
+            });
+            element.on('$destroy', function() {
+                element.off();
+            });
+
+
```
Comment (tbouron) [inline ui-modules/utils/catalog-uploader/catalog-uploader.js]: Couple of empty lines to remove

File: ui-modules/utils/catalog-uploader/catalog-uploader.js (hunk 1 above)
Comment (frogfather) [reply ui-modules/utils/catalog-uploader/catalog-uploader.js]: @tbouron  Strictly not, but it was a minor bug we corrected at the same time. Can make it into a separate PR if required.

Comment (tbouron) [reply ui-modules/utils/catalog-uploader/catalog-uploader.js]: @frogfather Leaving it there is fine but it would be great if you can explain why this is for

File: ui-modules/utils/catalog-uploader/catalog-uploader.js (hunk 2 above)
Comment (frogfather) [reply ui-modules/utils/catalog-uploader/catalog-uploader.js]: Done

File: ui-modules/utils/catalog-uploader/catalog-uploader.js (hunk 1 above)
Comment (frogfather) [reply ui-modules/utils/catalog-uploader/catalog-uploader.js]: @tbouron Added an explanation in the description of the PR. Do you also want comments in the code?
